### ISSUE TITLE
Unable to use a `union()` as a `record()` key value?

### ISSUE THREAD

**Issue by Saeris** [fabian-hiller/valibot#201]
Trying to port some code from Zod and noticed this pattern doesn't appear to translate as easily as I'd like.

```ts
const localesSchema = z.union([
  z.literal(`en-US`),
  z.literal(`en-GB`),
  z.literal(`es-ES`),
  z.literal(`de`),
  z.literal(`fr`),
  z.literal(`it`),
  z.literal(`zh-CN`)
]) // type Locales = "en-US" | "en-GB" | "es-ES" | "de" | "fr" | "it" | "zh-CN"

const nameLocalizationsSchema = z.record(
  localesSchema,
  z.string().min(1).max(100)
) // type NameLocalizations = Record<Locales, string>
```

But what I would assume to be the equivalent code in Valibot doesn't appear to be valid

```ts
const localesSchema = union([
  literal(`en-US`),
  literal(`en-GB`),
  literal(`es-ES`),
  literal(`de`),
  literal(`fr`),
  literal(`it`),
  literal(`zh-CN`)
])

const nameLocalizationsSchema = record(
  localesSchema, // Error, not assignable to RecordKey
  string([minLength(1), maxLength(100)])
)
```

While you could argue that an enum is a better fit here than a union, it's still perfectly valid Typescript to use a string union as a Record key type. Would appreciate it if Valibot could support this!

**Comment by fabian-hiller** [comment]
I will try to improve in the next version. For now I recommend to use `enumType` as a workaround:

```ts
const localesSchema = enumType([
  'en-US',
  'en-GB',
  'es-ES',
  'de',
  'fr',
  'it',
  'zh-CN',
]);

const nameLocalizationsSchema = record(
  localesSchema,
  string([minLength(1), maxLength(100)])
);
```

**Comment by Saeris** [comment]
Thanks that seems to work for now! Of course I'm reading through the source code to try and work out the API and usage information for all the functions so it wasn't clear how to do this migration.

**Comment by fabian-hiller** [comment]
We will try to provide a guide for migration in the future.

**Comment by fabian-hiller** [comment]
Fixed in v0.19.0
